src/lsstseries/ensemble.py

```diff
@@ -475,21 +479,46 @@ def _generate_object_table(self):
             .pivot_table(values=self._time_col, index=self._id_col, columns=self._band_col, aggfunc="sum")
         )
 
+        # If the ensemble's keep_empty_objects attribute is True and there are previous
+        # objects, then copy them into the res table with counts of zero.
+        if self.keep_empty_objects and self._object is not None:
+            # Check that there are existing object ids.
+            object_inds = self._object.index.values.compute()
+            if len(object_inds) > 0:
+                # Determine which object IDs are missing from the source table.
+                source_inds = self._source.index.values.compute()
+                missing_inds = np.setdiff1d(object_inds, source_inds).tolist()
+
+                # Create a dataframe of the missing IDs with zeros for all bands and counts.
+                rows = {self._id_col: missing_inds}
+                for i in res.columns.values:
+                    rows[i] = [0] * len(missing_inds)
+
+                zero_pdf = pd.DataFrame(rows, dtype=int).set_index(self._id_col)
+                zero_ddf = dd.from_pandas(zero_pdf, sort=True, npartitions=1)
+
+                # Concatonate the zero dataframe onto the results.
+                res = dd.concat([res, zero_ddf], interleave_partitions=True).astype(int)
```

Comment:
My only other comment is how this creates a new partition for the zero source objects. I _think_ the preferred option for now would be to append these to the last partition? My thinking being this is the object table, which is generally much smaller than the source table. In many cases I think users will only need a single partition for their object table, so appending them would preserve that. For cases where we need many partitions for the object table I think we would eventually have them being appended by division, where each index is slotted into an existing partition using the divisions as guidelines. But we're not there yet on that end. Just my thinking on this though.